### PR TITLE
Add workflow building for macOS arm64

### PR DESCRIPTION
**Pull request by jonatanklosko**
:cat2:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,7 +38,8 @@ $(TENSORFLOW_DIR):
 		git init && \
 		git remote add origin $(TENSORFLOW_GIT_REPO) && \
 		git fetch --depth 1 origin $(TENSORFLOW_GIT_REV) && \
-		git checkout FETCH_HEAD
+		git checkout FETCH_HEAD && \
+		rm $(TENSORFLOW_DIR)/.bazelversion
```
Comment (jonatanklosko) [inline Makefile:42]: I also verified that all build types pass with Bazel 4.2.1, so I think we could instead write "4.2.1" to this file and update the README to always require it. Will do so in a separate PR so the README stays accurate until the next relese.

**Review by josevalim — APPROVED** [review]
Beautiful!!!
